model-server/src/test/kotlin/org/modelix/model/server/handlers/ModelReplicationServerTest.kt

```diff
@@ -133,4 +144,59 @@ class ModelReplicationServerTest {
             assertEquals(HttpStatusCode.InternalServerError, response.status)
         }
     }
+
+    @Test
+    fun `server closes connection when failing to compute delta after starting to respond`() =
+        kotlinx.coroutines.test.runTest {
+            // Arrange
+            val repositoryId = RepositoryId("repo1")
+            val branchRef = repositoryId.getBranchReference()
+            val storeClient = InMemoryStoreClient()
+            val modelClient = LocalModelClient(storeClient)
+            val repositoriesManager = RepositoriesManager(modelClient)
+            val faultyRepositoriesManager = object :
+                IRepositoriesManager by repositoriesManager {
+                override suspend fun computeDelta(versionHash: String, baseVersionHash: String?): ObjectData {
+                    val originalFlow = repositoriesManager.computeDelta(versionHash, baseVersionHash).asFlow()
+                    val brokenFlow = channelFlow<Pair<String, String>> {
+                        error("Unexpected error.")
+                    }
+                    return ObjectDataFlow(
+                        flow {
+                            emitAll(originalFlow)
+                            emitAll(brokenFlow)
+                        },
+                    )
+                }
+            }
+            repositoriesManager.createRepository(repositoryId, null)
+
+            val modelReplicationServer = ModelReplicationServer(faultyRepositoriesManager, modelClient, InMemoryModels())
+            val setupBlock = { application: Application -> modelReplicationServer.init(application) }
+            val testBlock: suspend (server: NettyApplicationEngine) -> Unit = { server: NettyApplicationEngine ->
+                withTimeout(10.seconds) {
+                    val port = server.resolvedConnectors().first().port
+                    val url = "http://localhost:$port/v2"
+                    val client = HttpClient(CIO) {
+                        install(HttpTimeout) {
+                            requestTimeoutMillis = 5_000
+                        }
+                    }
```

Comment:
Couldn't the runWithNettyServer provide the client as the argument to the test block. That would make reading this test a bit easier.

Reply:
For now I do not know the futur usage patterns and would like to keep it flexibel.
For example I'm not sure, which timeout is reasonabel for the client.
Same goes for the timeout in the `testBlock` `withTimeout(10.seconds)`.

Reply:
Alright, so then maybe put the client creation code that also derives the URL etc into a method somewhere close to the test? I was pretty distracted from all the client creation details when trying to understand what is actually being tested here.

Reply:
Done. I created method for creating the client inside the test method.